opentelemetry-otlp/examples/basic-otlp-http/src/main.rs

```diff
@@ -4,10 +4,33 @@ use opentelemetry::{
     trace::{TraceContextExt, TraceError, Tracer},
     Key, KeyValue,
 };
+use opentelemetry_appender_tracing::layer::OpenTelemetryTracingBridge;
 use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::logs as sdklogs;
 use opentelemetry_sdk::metrics as sdkmetrics;
+use opentelemetry_sdk::resource;
 use opentelemetry_sdk::trace as sdktrace;
+
 use std::error::Error;
+use tracing::info;
+use tracing_subscriber::prelude::*;
+
+fn init_logs() -> Result<sdklogs::Logger, opentelemetry::logs::LogError> {
+    opentelemetry_otlp::new_pipeline()
+        .logging()
+        .with_log_config(
+            sdklogs::Config::default().with_resource(resource::Resource::new(vec![KeyValue::new(
+                opentelemetry_semantic_conventions::resource::SERVICE_NAME,
+                "basic-otlp-logging-example",
```

Comment:
Consider passing `env!("CARGO_BIN_NAME")` instead of hard-coding the example name, as that demonstrates best practice and works if someone copies the example and renames their binary.

Reply:
Good point, using hardcoded name now only if the env-variable is not set:
```rust
    let service_name = std::env::var("CARGO_BIN_NAME")
        .unwrap_or_else(|_| String::from("basic-otlp-http-logging-example"));
```